garden_ai/mlmodel.py

```diff
@@ -16,90 +17,150 @@ class ModelUploadException(Exception):
     pass
 
 
-def upload_model(
-    model_path: str,
-    model_name: str,
-    user_email: str,
-    flavor: str,
-    extra_pip_requirements: List[str] = None,
-) -> str:
+class DatasetConnection(BaseModel):
+    """
+    A first step towards the Accelerate Connection Schema
+    """
+
+    type: str = "dataset"
+    relationship: str = "origin"
+    doi: str = Field(...)
+    repository: str = "Foundry"
+    url: str = Field(...)
+
+
+class LocalModel(BaseModel):
+    """
+    The ``LocalModel`` class represents a trained ML model
+    that a user wants to register with Garden.
+
+    Args:
+    model_name (str) A short and descriptive name of the model
+    flavor (str): The framework used for this model. One of "sklearn", "tensorflow", or "torch".
+    extra_pip_requirements (List[str]), optional
+        A list of additional pip requirements needed to load and/or run the model.
+        Defaults to None.
+    local_path (str): Where the model is located on disk. Can be a file or a directory depending on the flavor.
+    user_email (str): The email address of the user uploading the model.
+
+    """
+
+    model_name: str = Field(...)
+    flavor: str = Field(...)
+    extra_pip_requirements: List[str] = Field(default_factory=list)
+    local_path: str = Field(...)
+    user_email: str = Field(...)
+    connections: List[DatasetConnection] = Field(default_factory=list)
+    namespaced_model_name: Optional[str]
```

Comment:
nit: while we're waiting for pydantic v2, looks like the preferred way to manipulate attributes on instances is to use `PrivateAttr` https://docs.pydantic.dev/latest/usage/models/#private-model-attributes

Reply:
I don't want it to be private though. I just want it to be a function of other attributes.

Reply:
yeah no you're totally correct I misunderstood -- we do want this attribute in the json dump of the metadata, right? not sure why I thought we didn't when I wrote that

Reply:
> we do want this attribute in the json dump of the metadata, right?

Yeah exactly